DEV-GUIDELINES.md

```diff
@@ -0,0 +1,187 @@
+Development Guidelines
+------------
+
+This document provides a list of rules and best practices followed by project Helidon.
+Please follow these rules when contributing to the project or when refactoring existing code.
+
+**There may be reasons to violate these guidelines - in such cases, this must be documented in code, as otherwise 
+    review will fail.**
```

Comment:
How about removing "review will fail" and simply mention in the first paragraph that these rules and meant to be used for code review.

Reply:
Modified the first paragraph